Fixing the reduce documentation (#431)
rescript-association/rescript-lang.org: pages/docs/manual/latest/api/belt/array.mdx

let reduce: (array<'b>, 'a, ('a, 'b) => 'a) => 'a
```

`reduce(arr, init, f)`

Applies `f` to each element of `arr`.

> Function `f` has two parameters: an "accumulator" which starts with a value of `init` and the next `value` from the array. 

Reviewer: ```suggestion
Function `f` has two parameters: an "accumulator" which starts with a value of `init` and the next value from the array. 
```

let reduce: (array<'b>, 'a, ('a, 'b) => 'a) => 'a
```

`reduce(arr, init, f)`

Applies `f` to each element of `arr`.

Function `f` has two parameters: an "accumulator" which starts with a value of `init` and the next `value` from the array. 

It returns the final value of the accumulator.

```res example

> Belt.Array.reduce([2, 3, 4], 1, (a, b) => a + b) == 10

Reviewer: ```suggestion
Belt.Array.reduce([2, 3, 4], 1, (acc, value) => acc + value) == 10
```

> `reduce(xs, init, f)`

Reviewer: ```suggestion
`reduce(arr, init, f)`
```

> Applies `f` to each element of `xs` from beginning to end. Function `f` has two parameters: an “accumulator” ; which starts with a value of `init` and the item from the list. `reduce` returns the final value of the accumulator.

Reviewer: ```suggestion
Applies `f` to each element of `arr`.

Function `f` has two parameters: an "accumulator" which starts with a value of `init` and the next `value` from the array. 

It returns the final value of the accumulator.
```
